### ISSUE TITLE
Cannot setup Global Accent Color for macOS Big Sur

### ISSUE THREAD

**Issue by ivanicin** [xamarin/xamarin-macios#10275]
### Steps to Reproduce

On macOS Big Sur you can set Global Accent Color in Xcode project properties.

### Expected Behavior
This should be possible in Xamarin too. I believe that in Xamarin style this should be a custom Info.plist entry. However it is not clear and even if it exists it should be pre-filled and also a sample accent color should be added to the assets catalog, similar to what Xcode does.

### Actual Behavior
As far as I have checked this is not possible in Xamarin, or at least it is very undiscoverable and not documented

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.8.3 (build 16)
Installation UUID: 1b954348-4e6f-4675-9727-14f156acf931
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000107

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.107 (2020-02/a22ed3f094e) (64-bit)
	Package version: 612000107

=== Roslyn (Language Service) ===

3.8.0-5.20519.18+4c195c3ac1974edcefa76774d7a59a2350ec55fa

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.100/Sdks
SDK Versions:
	5.0.100
	3.1.404
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	2.1.701
	2.1.700
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.0
	3.1.10
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.12
	2.1.11

=== .NET Core 3.1 SDK ===

SDK: 3.1.404

=== Xamarin.Profiler ===

Version: 1.6.12.26
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 11.1.0.17 (Visual Studio Community)
Commit: xamarin-android/d16-8/c0e2b8e
Android SDK: /Users/ivanicin/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 28.0.3

Build Information: 
Mono: be2226b
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/ivanicin/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:30 UTC

=== Android Device Manager ===

Version: 16.8.0.45
Hash: fc0af5f
Branch: remotes/origin/d16-8
Build date: 2020-10-13 23:32:54 UTC

=== Apple Developer Tools ===

Xcode 12.2 (17535)
Build 12B45b

=== Xamarin.Mac ===

Version: 7.0.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:40-0500

=== Xamarin.iOS ===

Version: 14.6.0.15 (Visual Studio Community)
Hash: 87a1b18d8
Branch: d16-8
Build date: 2020-11-16 21:39:41-0500

=== Xamarin Designer ===

Version: 16.8.0.507
Hash: e87b24884
Branch: remotes/origin/d16-8
Build date: 2020-10-29 00:31:38 UTC

=== Build Information ===

Release ID: 808030016
Git revision: cc6375d2fdcc1b42532e551e24ce650eaca1ae45
Build date: 2020-11-25 13:16:20-05
Build branch: release-8.8
Xamarin extensions: cc6375d2fdcc1b42532e551e24ce650eaca1ae45

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.1.0 Darwin Kernel Version 20.1.0
    Sat Oct 31 00:07:11 PDT 2020
    root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64


```

**Comment by ivanicin** [comment]
@rolfbjarne while it is likely that you are right, I would like to check, is this issue really external? As far as I can tell you have resolved the selection of Icon asset that should be essentially the same problem by adding the entry to plist, in which case I think there is nothing external to this, just the compiler should do the thing when it compiles. And then maybe this is external too, but my best guess is that it is not... so... just asking :).

**Comment by rolfbjarne** [comment]
@ivanicin by the external-* labels we just mean that this is for a different team in Microsoft (basically this is something that should be implemented in the IDEs).

**Comment by ivanicin** [comment]
@rolfbjarne @chamons  it has been almost half the year since the issue has been submitted.

While I do understand that Xamarin doesn't intend to support everything on Xamarin.Mac I would consider this is as something basic and essential - equivalents are supported by Xamarin whenever they do exist on any other platform. If you don't support essential things like this it leaves the impression that Xamarin.Mac may be left without the support at some point. And I am quite certain that if you don't take this in the next month or two it won't be scheduled until the next year as you'll be chasing whatever is announced on WWDC at that point (which is fair but it is not fair that this is not implemented).

Any chance for you to boost the priority of this?

**Comment by dalexsoto** [comment]
Hello

Thanks for checking in, I'll talk to the IDE folks and reply back to you, that said I want to set some expectations right now the whole SDKs, MAUI and IDE teams are going through a heavy transition to .NET 6 and Xamarin.Mac is part of this transition so it is definitely not going away and it is here to stay.

**Comment by ivanicin** [comment]
> Thanks for checking in, I'll talk to the IDE folks and reply back to you, that said I want to set some expectations right now the whole SDKs, MAUI and IDE teams are going through a heavy transition to .NET 6 and Xamarin.Mac is part of this transition so it is definitely not going away and it is here to stay.

@dalexsoto  thank you for your answer.

First I would like to say that I understand that most people in the Xamarin team go through a very stressful period of those transitions. And while this answer may add up to that stress, my intention is to help you to observe things better and eventual get to the better situation.

I appreciate your statement that Xamarin.Mac will continue to be supported, but I would like to stress that at this point it is not supported at all. Here is why:

- current support doesn't go beyond few guys are checking now and then some problems and may even do sometimes something, though likely they will not no matter how big and urgent the problem is. This is literary my experience, and I have followed the project for 5 years and I couldn't say that before
- major functionalities that were implemented for years and used in apps are broken for 6 months and it doesn't seem that anyone cares (this relates to the share extension in macOS that doesn't work for 6 months, and for several months no extension seemed to work though at least that is fixed)
- a simple things like wrong bound APIs that need few hours of work to be fixed and may have notable implications are not worked on for months. I've provided the description that could be tested in less than a minute and I was met with answer that a sample project is required. When I said that it is not required but that I will do even that if someone can confirm that the bug will be worked on in the next few months I've received no reply. 
- finally some expected improvements in catching up with major Apple's developments are far off.

Certainly I would accept your excuse that you go through MAUI and .NET 6 transition if it was a one time excuse. But good developers know that such things don't come out of nothing and are a sign of deeper problems within the project.

First thing to note is that you have decided to go through two major transitions at a same time, something that is likely risky by itself and should require a lot of thinking before accepting.

Second thing to note is that you came at the situation where even the most essential support is not provided and you don't seem to find it urgent and particularly wrong. I mean I am quite sure that you will confirm at the philosophical level that it is wrong, but it will cause no action from your side which means that you still don't find it particularly wrong.

Finally, there is always some exit, but you don't search for it. You can still postpone MAUI if that is the only solution, my personal opinion is that it is very nice, but nothing revolutionary. Nothing that is worth risking the basic credibility of your team. Also Microsoft is a big company, you may pool people from other teams or get some new employees for a year or two to handle at least some basic tasks (fixing of wrongly bound APIs doesn't appear to be the most complex thing ever).

I cannot say what is the source of your problem as I don't have any inside information. I am quite certain that Miguel de Icaza wouldn't allow something like this while he was there, so at least in part it is a leadership problem.

I must say that I write this because I have invested a lot in some Xamarin solution and I wouldn't like to be forced to rewrite apps. So I am not writing like an enemy but rather someone who is interested that this project has a long-term viability.

**Comment by iainx** [comment]
I was looking at this recently, and this isn't really a vsmac/vs-extensions bug. To set the accent colour you need to pass an option to actool when compiling the asset catalog specifying the accent colour: `--accent-color <name>`. This would need to be set in https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ACToolTaskBase.cs

 The standard way this is done is having a default colourset in the assets catalog called AccentColor. An improvement to vs-mac-extensions would be to add this default colourset in the template asset catalog, but unless xamarin.mac adds the --accent-color option to actool, it won't do anything.

**Comment by Therzok** [comment]
One workaround that can be done is overriding the ACTool task and use that instead, where you can override `AppendCommandLineArguments`, but it seems like a big hacky approach.

The Task should offer a way to provide additional commandline arguments via MSBuild, so users aren't stuck waiting for an SDK change to do it.

**Comment by Therzok** [comment]
You can add a [UsingTask](https://docs.microsoft.com/en-us/visualstudio/msbuild/usingtask-element-msbuild?view=vs-2019) that points to your subclassed new task that also appends `--accent-color`, as a workaround until this is fixed. This must be done before importing Xamarin.Mac props.

**Comment by ivanicin** [comment]
> 
> 
> You can add a [UsingTask](https://docs.microsoft.com/en-us/visualstudio/msbuild/usingtask-element-msbuild?view=vs-2019) that points to your subclassed new task that also appends `--accent-color`, as a workaround until this is fixed. This must be done before importing Xamarin.Mac props.

@Therzok thanks, this sounds very promising, but while developers here surely have higher level of knowledge than average I would say that this is a bit advanced topic for most so it would be great if you could elaborate with few more details. I'll surely implement the workaround as soon as I understand what I should do to make it work. 

Also big thanks for @iainx for locating the problem.

**Comment by Therzok** [comment]
@ivanicin MSBuild has support for loading assemblies with custom tasks implemented, similarly to how `Xamarin.Mac.Build.Tasks.dll` does.

What you would normally do is create a project that references MSBuild assemblies and the Xamarin.Mac tasks assembly, subclass [ACTool](https://github.com/xamarin/xamarin-macios/blob/3d9c37778f0ea81fdb02ce16ed0ecefb9cc13152/msbuild/Xamarin.Mac.Tasks/Tasks/ACTool.cs#L5) and override `AppendCommandLineParameters` to also add `--accent-color`.

**Comment by ivanicin** [comment]
> @ivanicin MSBuild has support for loading assemblies with custom tasks implemented, similarly to how `Xamarin.Mac.Build.Tasks.dll` does.
> 
> What you would normally do is create a project that references MSBuild assemblies and the Xamarin.Mac tasks assembly, subclass [ACTool](https://github.com/xamarin/xamarin-macios/blob/3d9c37778f0ea81fdb02ce16ed0ecefb9cc13152/msbuild/Xamarin.Mac.Tasks/Tasks/ACTool.cs#L5) and override `AppendCommandLineParameters` to also add `--accent-color`.

Just to note that I have ran the bounty on StackOverflow to check if I am correct that while this is a welcome explanation it is at the level that probably less than 1% of developers have: https://stackoverflow.com/questions/67961343/setting-accent-color-on-xamarin-mac-big-sur

It is reasonable to assume that the Microsoft official support that is active and helpful on the site had to check this without being able to answer as no one answered.

I'll also see to offer the paid project on Freelancer after this, but I am not hopeful that it will deliver any better results, developers with so extremely high quality skills are rarely available there. I'll report on that when I receive the feedback, but for now it looks that some blog post on this with more details on how to make this operational (or a simple sample project on the github) would be highly welcome. Unless this is planned to be quickly fixed, which doesn't appear to be the case.

**Comment by ivmirx** [comment]
I agree with @ivanicin – Xamarin.Mac feels abandoned. I decided to go with it anyway for the next project since most of AppKit support is robust, but it's disappointing when tiny things like the accent color are not being added for almost a year after the new OS release.

Sure, even the native Mac app development scene is tiny, and probably only a few hundred people at best use Xamarin.Mac, but Apple themselves add very little to AppKit. Not having new features in place when a major macOS version is released for the general public threatens the future of those few projects who picked .NET for desktop cross-platform apps, and it fuels the vicious cycle where even fewer developers risk going this route, so fewer Xamarin.Mac apps, etc.

Regarding the issue, the solution is indeed too advanced for me because I'm not familiar with MSBuild. I will look into it closer to my app's release, but it would be great to have most long-standing Mac issues closed when .NET 6 is out and you have less workload.

**Comment by ivanicin** [comment]
@ivmirx as no one wrote a sample for this ms-build workaround from Microsoft which likely doesn't take more than 15 minutes to explain here in details, you can conclude that either no one knows or no one cares. I am not aware of the 3rd possible answer.

**Comment by ivmirx** [comment]
@ivanicin to me it seems easier to make a PR to [the class](https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ACToolTaskBase.cs) @iainx linked because [the code that processes](https://github.com/xamarin/xamarin-macios/blob/ecad9d96676d1be7241c3548bc5fb9708cb5d2aa/msbuild/Xamarin.MacDev.Tasks.Core/Tasks/ACToolTaskBase.cs#L92) the app's icon is quite straightforward and passing the accent color shouldn't be too different.

**Comment by ivanicin** [comment]
@dalexsoto 
> Hello
> 
> Thanks for checking in, I'll talk to the IDE folks and reply back to you, that said I want to set some expectations right now the whole SDKs, MAUI and IDE teams are going through a heavy transition to .NET 6 and Xamarin.Mac is part of this transition so it is definitely not going away and it is here to stay.

Now that .NET 6 is out, possibly you can take a look at this? Maybe two years after the Mac developers we can get this essential UI support?

**Comment by ivanicin** [comment]
@rolfbjarne you now (when this gets released) need just an entry named AccentColor in the Assets to achieve the effect?

**Comment by rolfbjarne** [comment]
@ivanicin you'll also need to add this to the project file:

```xml
<PropertyGroup>
    <AccentColor>AccentColor</AccentColor>
</PropertyGroup>
```

**Comment by ivanicin** [comment]
OK, I'll start the test today.
